From fbd7160e53b11d09f845b0bfcaeb5e41fa958bd2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Sat, 20 Jul 2024 14:34:05 +0000
Subject: [PATCH] validation: always call lemon with -l and relative path to
 source

lemon supports -l since 2008:
https://github.com/sqlite/sqlite/commit/5854393cd9d3ca956669673ff249ef999f75c056

even the lemon from sqlite 3.7.3 used before the upgrade in:
https://git.openembedded.org/meta-openembedded/commit/?id=c3ac5cf180f960dbbf6e3a3e76d6d44094873c72
did support -l

So not sure why -l<file> was added to apply_lemon.sh in 2015 with:
https://g2g.lgsvl.com/7102 Add support for lemon extra arguments
  http://gpro.lge.com/gitweb?p=webos/libpbnjson.git;a=commit;h=78331aab6435b41fd3d133fa92c66accdb95eecd
https://g2g.lgsvl.com/7181 Strip #line from autogenerated file
  http://gpro.lge.com/gitweb?p=webos/libpbnjson.git;a=commit;h=d455a73f29affda4290aad833ff5616ab043a1a5

instead of just adding -l to lemon call.

The sed worked but we should also call lemon with relative path to grammar file, because
it also generates the buildpath as "source file" in comment as:

** source file "/OE/build/luneos-styhead/tmp-glibc/work/corei7-64-webos-linux/libpbnjson/2.15.0-16/build/src/pbnjson_c/validation/json_schema_grammar.y".

Added in 2020 sqlite 3.45.1:
https://github.com/sqlite/sqlite/commit/512aa78ca28981229858e0fd98aaba1158b6e8af

and it's still in latest sqlite:
https://github.com/sqlite/sqlite/commit/4fa5952090ad974d6a11952267662d4af338154a#diff-633c611978778713c1360ca3046e277e54c6a7569108c74b3300da07133d1933R4412

without easy way to strip the path.

I don't think we need the full path, so just cd into output dir before
lemon call and use just the filename.

Fixes:
ERROR: libpbnjson-2.15.0-16-r17 do_package_qa: QA Issue: File /usr/src/debug/libpbnjson/2.15.0-16/src/pbnjson_c/validation/json_schema_grammar.c in package libpbnjson-src contains reference to TMPDIR [buildpaths]

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Pending [reported in WRP-10884 still waiting for component owner to work]

 src/pbnjson_c/validation/CMakeLists.txt |  5 -----
 src/pbnjson_c/validation/apply_lemon.sh | 19 ++-----------------
 2 files changed, 2 insertions(+), 22 deletions(-)

diff --git a/src/pbnjson_c/validation/CMakeLists.txt b/src/pbnjson_c/validation/CMakeLists.txt
index b592307..11f4186 100644
--- a/src/pbnjson_c/validation/CMakeLists.txt
+++ b/src/pbnjson_c/validation/CMakeLists.txt
@@ -17,10 +17,6 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 
-set(LEMON_WITH_LINE TRUE CACHE BOOL "Strip lemon generated #line statements")
-set(LEMON_WITH_LINE_TRUE_ARGS "")
-set(LEMON_WITH_LINE_FALSE_ARGS "-l${CMAKE_CURRENT_BINARY_DIR}/json_schema_grammar.c")
-
 add_custom_target(
 	json_schema_grammar echo "Creating json_schema_grammar.c"
 	)
@@ -28,7 +24,6 @@ add_custom_target(
 add_custom_command(
 	SOURCE json_schema_grammar.y
 	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/apply_lemon.sh
-		${LEMON_WITH_LINE_${LEMON_WITH_LINE}_ARGS}
 		${LEMON}
 		${CMAKE_CURRENT_BINARY_DIR}
 		${CMAKE_CURRENT_SOURCE_DIR}
diff --git a/src/pbnjson_c/validation/apply_lemon.sh b/src/pbnjson_c/validation/apply_lemon.sh
index f144765..680f678 100755
--- a/src/pbnjson_c/validation/apply_lemon.sh
+++ b/src/pbnjson_c/validation/apply_lemon.sh
@@ -16,19 +16,6 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-STRIP_LINES=false
-STRIP_FILE=
-
-OPTIND=1
-while getopts l: opt; do
-	case "$opt" in
-		l) STRIP_LINES=true
-		   STRIP_FILE="$OPTARG" ;;
-	esac
-done
-shift $((OPTIND-1))
-[[ "$1" = '--' ]] && shift
-
 lemon="$1"; shift
 binary_dir="$(readlink -f "$1")"; shift
 source_dir="$(readlink -f "$1")"; shift
@@ -60,7 +47,5 @@ if [ "$source_dir" != "$binary_dir" ]; then
 		cp -af "$source_dir/$file" "$binary_dir/" || \
 		exit 1
 fi
-"$lemon" "$@" "$binary_dir/$file" || exit 1
-if [ "$STRIP_LINES" = "true" ]; then
-	sed -i -e '/^#line/d' "$STRIP_FILE"
-fi
+cd $binary_dir
+"$lemon" "$@" "$file" || exit 1
